src/Microsoft.Health.SqlServer/Registration/SqlServerBaseRegistrationExtensions.cs

```diff
@@ -81,7 +81,19 @@ public static IServiceCollection AddSqlServerConnection(this IServiceCollection
 
             // The following are only used in case of managed identity
             services.AddSingleton<IAccessTokenHandler, ManagedIdentityAccessTokenHandler>();
-            services.AddSingleton<AzureServiceTokenProvider>();
+            services.AddSingleton<AzureServiceTokenProvider>(p =>
```

Comment:
Not sure it's any easier, but it appears that `Microsoft.Data.SqlClient` has some support for just saying use managed identity in the connection string. https://docs.microsoft.com/en-us/sql/connect/ado-net/sql/azure-active-directory-authentication?view=sql-server-ver15#using-active-directory-managed-identity-authentication

Reply:
I think this will require reprovisioning all existing accounts